jni/jni.cabal

```diff
@@ -18,15 +18,20 @@ source-repository head
   location: https://github.com/tweag/inline-java
   subdir: jni
 
+flag ghc-head
```

Comment:
Using `impl(ghc > 8.9)` instead of a flag, would probably cause the package to not build when ghc-8.9 is finally released.

The flag makes it more evident to users and build tools that the code we have here is only a transient solution for testing.

inline-java.cabal uses `impl(ghc >= 8.9)` because the code there is likely to remain valid.